src/Corporation/Actions.ts

```diff
@@ -198,125 +198,120 @@ export function acceptInvestmentOffer(corporation: Corporation): void {
   corporation.investorShares += investShares;
 }
 
-export function sellMaterial(material: Material, amount: string, price: string): void {
-  if (price === "") price = "0";
-  if (amount === "") amount = "0";
-  let cost = price.replace(/\s+/g, "");
-  cost = cost.replace(/[^-()\d/*+.MPe]/g, ""); //Sanitize cost
-  let temp = cost.replace(/MP/, "1.234e5");
+export function convertPriceString(price: string): string | number {
+  // Replace whitespace characters.
+  let sanitizedPrice = price.replace(/\s+/g, "");
+  /**
+   * Replace invalid characters. Only accepts:
+   * - Digit characters
+   * - 4 most basic algebraic operations (+ - * /)
+   * - Parentheses
+   * - Dot character
+   * - e
+   * - MP
+   */
+  sanitizedPrice = sanitizedPrice.replace(/[^\d+\-*/().eMP]/g, "");
+
+  // // If the input does not contain "MP", we only need to convert it to a number.
```

Comment:
This whole block seems needlessly complicated to me. We're going to eval the string later; why are we going to so much effort to avoid doing so now?

The only reason we (re-do) all this work is to give the user a decent error message up-front, when they type in the box. As a result, the check here should mirror the actual calculation as much as possible. Which means, I think we can just eliminate this whole block.